datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala

```diff
@@ -0,0 +1,33 @@
+package za.co.absa.hermes.datasetComparison
+
+import org.apache.commons.cli.MissingArgumentException
+import org.apache.spark.sql._
+
+case class DataframeOptions(format: String, options: Map[String, String], path: String) {
+  def loadDataFrame(implicit spark: SparkSession): DataFrame = {
+    val general = spark.read.format(format)
+    val generalWithOptions = if ( options.isEmpty ) general else general.options(options)
+
+    if (format == "jdbc") { generalWithOptions.load()     }
+    else                  { generalWithOptions.load(path) }
+  }
+}
```

Comment:
Maybe 2 more methods:

`private def setOptions(reader: DataReader, options: Map[String,String]): DataReader`

and 

`private def load(path: String, reader: DataReader): DataFrame`

then, the whole thing would be:

```
val general = spark.read.format(format)
val withOptions = setOptions(general, options)
load(path, withOptions)
```

Reply:
Done